api/envoy/config/listener/v3/listener_components.proto

```diff
@@ -109,6 +110,163 @@ message FilterChainMatch {
     EXTERNAL = 2;
   }
 
+  // Matches filter chains by the destination port.
+  message DestinationPortInput {
+  }
+
+  // Matches filter chains by the destination IP address.
+  message DestinationIPInput {
+  }
+
+  // Matches filter chains by the source IP address.
+  message SourceIPInput {
+  }
+
+  // Matches filter chains by the directly connected source IP address (this
+  // will only be different from the source IP address when using a listener
+  // filter that overrides the source address, such as the :ref:`Proxy Protocol
+  // listener filter <config_listener_filters_proxy_protocol>`).
+  message DirectSourceIPInput {
+  }
+
+  // Matches filter chains by the source port.
+  message SourcePortInput {
+  }
+
+  // Matches filter chains by the connection source IP type.
+  // Specifies the connection source IP match type. The values include:
+  //
+  // * ``any`` - any connection source,
+  // * ``local`` - matches a connection originating from the same host,
+  // * ``external`` - matches a connection originating from a different host
+  message SourceTypeInput {
+  }
+
+  // Matches filter chains by the server names of a new connection, when detected by one of the listener filters.
+  message SNIInput {
+  }
+
+  // Matches filter chains by the transport protocol.
+  // This value will be compared against the transport protocol of a new connection, when
+  // it's detected by one of the listener filters.
+  //
+  // Suggested values include:
+  //
+  // * ``raw_buffer`` - default, used when no transport protocol is detected,
+  // * ``tls`` - set by :ref:`envoy.filters.listener.tls_inspector <config_listener_filters_tls_inspector>`
+  //   when TLS protocol is detected.
+  message TransportProtocolInput {
+  }
+
+  // Matches filter chains by the application protocol (e.g. ALPN for TLS protocol).
+  // This values will be compared against the application
+  // protocols of a new connection, when detected by one of the listener filters.
+  //
+  // Suggested values include:
+  //
+  // * ``http/1.1`` - set by :ref:`envoy.filters.listener.tls_inspector
```

Comment:
should we mention this is set by http_inspector as well?

Reply:
Updated.